Dockerfile-cni-plugin

```diff
@@ -0,0 +1,40 @@
+# syntax=docker/dockerfile:1.4
+
+ARG BUILDPLATFORM=linux/amd64
+
+# Precompile key slow-to-build dependencies
+FROM --platform=$BUILDPLATFORM golang:1.18-alpine as golang
+WORKDIR /linkerd-build
+COPY --link go.mod go.sum .
+COPY --link ./cni-plugin ./cni-plugin
+COPY --link ./proxy-init ./proxy-init
```

Comment:
Welll... probably not the best place to put this comment but we are building and using `proxy-init` only to use the `RootCommand` struct. We could perhaps  build `iptables.FirewallConfiguration` ourselves, or find another way to pass the options in so we can avoid coupling the CNI plugin and proxy-init together (outside of their shared `internal` package).

Reply:
We also use `BuildFirewallCommand` but I agree with you that can be pulled out since `BuildFirewallCommand` returns and `*iptables.FirewallConfiguration`. What do you think about putting it in a card and we can tackle it in a few weeks?

Reply:
A card sounds good 👌🏻